File: arrayLeftRotation.js
function leftRotate(size, num) {
  var array = [];
  var temp

  for (var i = 1; i <= size; i++) {
    array.push(i);
  }

  temp = array.splice(0,num)

  var rotate_array = array.concat(temp)
  return rotate_array

}

console.log(leftRotate(5,2))

File: arrayLeftRotation.rb
n,k = gets.strip.split(' ')
n = n.to_i
k = k.to_i
a = gets.strip
a = a.split(' ').map{|num| num.to_i}

def left_rotate(n, k, a)
  temp = a.slice!(0,k)
  rotated_array = a.concat(temp)
  return rotated_array
end

left_rotate(n,k,a)
